main/src/main/java/org/opennms/oce/tools/main/CpnOnmsSyslogAudit.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.tools.main;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.kohsuke.args4j.Option;
+import org.opennms.netmgt.syslogd.ByteBufferParser;
+import org.opennms.netmgt.syslogd.RadixTreeSyslogParser;
+import org.opennms.netmgt.syslogd.SyslogMessage;
+import org.opennms.oce.tools.cpn.EventUtils;
+import org.opennms.oce.tools.cpn.model.EventRecord;
+
+import com.google.gson.Gson;
+
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchResult;
+import io.searchbox.core.SearchScroll;
+import io.searchbox.core.search.sort.Sort;
+import io.searchbox.params.Parameters;
+
+public class CpnOnmsSyslogAudit extends AbstractCommand {
+    public static final String NAME = "cpn-opennms-syslog-audit";
+
+    @Option(name = "--from", aliases = {"-f"}, usage = "From date i.e. Oct 28 2018")
+    private String from;
+
+    @Option(name = "--to", aliases = {"-t"}, usage = "To date i.e. Oct 29 2018")
+    private String to;
+
+    @Option(name = "--count", aliases = {"-c"}, usage = "Set to print only the count")
+    private boolean countOnly = false;
+
+    private final ByteBufferParser<SyslogMessage> parser = RadixTreeSyslogParser.getRadixParser();
+
+    private long failedCount;
+    
+    private long processedCount;
+
+    public CpnOnmsSyslogAudit() {
+        super(NAME);
+    }
+
+    @Override
+    public void doExec(Context context) throws Exception {
+        scroll(context.getEsClient().getJestClient(), buildSearch(), EventRecord.class, this::consumeSyslogBatch);
```

Comment:
Can we use the ESDataProvider#getSyslogRecordsInRange instead? If not, can we at least move this logic to the ESDataProvider class.